src/RouteResolver.tsx

```diff
@@ -1,107 +1,150 @@
-  // This isn't typical, but just so we can load something, we (recursively) get the first child until we find a route with a SynapseConfigArray
-  while (isEmpty(route.synapseConfigArray) && route.isNested) {
-    route = route.routes[0]
-    routePathName += '/' + route.to
-  }
-  return [route, routePathName]
+  }, 500)
+  return (
+    <React.Fragment key={JSON.stringify(props)}>
+      {isOutsideContainer ? (
+        <div className={containerClassName}>
+          {title && (
+            <h2 className={`title ${centerTitle ? 'center-title' : ''}`}>
+              {title}
+            </h2>
+          )}
+          {subtitle && (
+            <p className={`${centerTitle ? 'center-title' : ''}`}>{subtitle}</p>
+          )}
+          <SynapseComponent
+            synapseConfig={props.config}
+            searchParams={searchParamsProps}
+          />
+        </div>
+      ) : (
+        <Layout
+          key={JSON.stringify(props)}
+          containerClassName={containerClassName}
+          outsideContainerClassName={outsideContainerClassName}
+        >
+          {scrollToJsx}
+          {/* re-think how this renders! remove specific styling */}
+          {title && (
+            <h2 className={`title ${centerTitle ? 'center-title' : ''}`}>
+              {title}
+            </h2>
+          )}
+          {subtitle && (
+            <p className={`${centerTitle ? 'center-title' : ''}`}>{subtitle}</p>
+          )}
+          <SynapseComponent
+            synapseConfig={props.config}
+            searchParams={searchParamsProps}
+          />
+        </Layout>
+      )}
+    </React.Fragment>
+  )
 }
 
-type SynapseComponentProps = {
-  synapseConfig: SynapseConfig
-}
+function RecursiveRouteRenderer(props: {
+  route: GenericRoute
+  getPageNameFromParentRoute: () => string
+}) {
+  const { route, getPageNameFromParentRoute } = props
+  const { url } = useRouteMatch()
 
-export const SynapseComponent: React.FC<SynapseComponentProps> = ({
-  synapseConfig,
-}) => {
-  const Component =
-    (PortalComponents as any)[synapseConfig.name] ??
-    (SynapseComponents as any)[synapseConfig.name]
-  if (!Component) {
-    throw Error(`No synapse object could be mapped for ${synapseConfig.name}`)
-  }
-  const component = <Component {...synapseConfig.props} />
-  const { style, className } = synapseConfig
-  if (style || className) {
-    return (
-      <div className={className} style={style}>
-        {component}
-      </div>
-    )
-  } else {
-    return component
+  const pageName = route.displayName ?? route.path?.replaceAll('/', '')
+  const getPageName = () =>
+    pageName
+      ? `${docTitleConfig.name} - ${pageName}`
+      : getPageNameFromParentRoute()
+
+  // if there are children, don't update the title
+  if (!route.routes || route.routes.length === 0) {
+    const newTitle: string = getPageName()
+    if (document.title !== newTitle) {
+      document.title = newTitle
+    }
   }
```

Comment:
Defer updating the document title until we render the most specific match in the tree. If that match doesn't have an appropriate title, fall back to the parent's title.